openapi_tester/schema_tester.py

```diff
@@ -89,11 +90,16 @@ def __init__(
             raise ImproperlyConfigured(INIT_ERROR)
 
     @staticmethod
-    def get_key_value(schema: dict[str, dict], key: str, error_addon: str = "") -> dict:
+    def get_key_value(schema: dict[str, dict], key: str, error_addon: str = "", use_regex=False) -> dict:
         """
         Returns the value of a given key
         """
         try:
+            if use_regex:
+                compiled_pattern = re.compile(key)
+                for key_ in schema.keys():
+                    if compiled_pattern.match(key_):
+                        return schema[key_]
```

Comment:
Why are we iterating over `key in schema.keys()` here instead of just doing this?

```python
if compiled_pattern.match(key):
    return schema[key_]
```

Reply:
This is to find a str match in the schema list. 
For example `schema.keys()` in one instance has `dict_keys(['application/xml', 'application/json'])` and in this case `key_` will return `schema["application/json"]` as opposed to `schema["^application\/.*json$"]` (which will break).